### ISSUE TITLE
run script mysteriously fails when github API rate limit is hit

### ISSUE THREAD

**Issue by jamezilla** [Brightify/Cuckoo#274]
This command in the [run script](https://github.com/Brightify/Cuckoo/blob/master/run#L15) silently eats the github API error when the rate limit is exceeded:
```
  DOWNLOAD_URL=$(curl ${REQUEST_HEADER+"$REQUEST_HEADER"} "$1" | grep -oe '"browser_download_url":\s*"[^" ]*"' | grep -oe 'http[^" ]*' | grep "$FILE_NAME" | head -1)
```
The log just shows the `DOWNLOAD_URL` variable is empty. This was very tedious to debug. It would be helpful to catch this error and log it so it's painfully obvious why it's failing.

**Comment by jamezilla** [comment]
A couple of possible approaches:
https://stackoverflow.com/questions/1550933/catching-error-codes-in-a-shell-pipe/4959616
https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail/

It's tempting to just throw a `set -eo pipefail` in there, but that I'm not sure whether a rate limit response causes a non-zero `curl` exit code (it's hard to reproduce without abusing github's API).

**Comment by MatyasKriz** [comment]
Just tried it by brute forcing with `curl` (it's only 60 requests per hour when not authorized) and can confirm that when the rate limit is exceeded, the exit code is still zero, just the rest of the pipe stops working.

The second `curl` however, the one that's there to download the actual binary, does fail because the `DOWNLOAD_URL` variable is empty. Maybe we could use that to tell the user about the exceeded GitHub API rate.

I'm not a fan of just exiting with the error `curl` gives as I imagine is what happens when using the command you're proposing. I'd instead go for the check for non-zero exit code of the second `curl` and `echo` an error message about the rate limit and exit with non-zero code.

**Comment by MatyasKriz** [comment]
Merged!

**Comment by MatyasKriz** [comment]
I added a check for the empty `DOWNLOAD_URL` variable, so it should be somewhat fixed in the upcoming `1.0.5`.
